docs/spec/sources.md

```diff
@@ -362,5 +376,39 @@ _Structured Content Mode_ of the HTTP Protocol Binding for CloudEvents. Sources
 SHOULD send events to its
 [Destination](https://pkg.go.dev/github.com/knative/pkg/apis/duck/v1?tab=doc#Destination).
 
+### CloudEvent `knsource` extension
+
+Sources SHOULD produce CloudEvents with the `knsource` CloudEvent extension
+identifying the originating [Source Custom Object](#source-custom-objects).
+
+The `knsource` value SHOULD be of the form `<name>.<plural>.<group>`, where:
+
+- `name` is the [Source Custom Object](#source-custom-objects) name
+- `plural` is the [Source CRD](#source-crds)
+  [plural](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)
+- `group` is the [Source CRD](#source-crds)
+  [API group](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)
```

Comment:
I am guessing this is a step to enable consuming events from a specific source, are these with the source field enough to identify a source? eg same name but in different namespace or even cluster.

Reply:
`with the source field`: what do you mean?